src/tmpfiles/tmpfiles.c

```diff
@@ -1785,6 +1789,52 @@ static bool should_include_path(const char *path) {
         return false;
 }
 
+static int specifier_expansion_from_arg(Item *i) {
+        _cleanup_free_ char *resolved = NULL;
+        char **name, **value;
+
+        int r;
+
+        assert(i);
+
+        if (i->argument == NULL)
+                return 0;
+
+        switch (i->type) {
+        case COPY_FILES:
+        case CREATE_SYMLINK:
+        case CREATE_FILE:
+        case TRUNCATE_FILE:
+        case WRITE_FILE:
+                r = specifier_printf(i->argument, specifier_table, NULL, &resolved);
+                if (r < 0)
+                        return r;
+                free_and_replace(i->argument, resolved);
+                break;
+
+        case SET_XATTR:
+        case RECURSIVE_SET_XATTR:
+                assert(i->xattrs);
+
+                STRV_FOREACH_PAIR (name, value, i->xattrs) {
+                        r = specifier_printf(*name, specifier_table, NULL, &resolved);
+                        if (r < 0)
+                                return r;
+                        free_and_replace(*name, resolved);
+
+                        r = specifier_printf(*value, specifier_table, NULL, &resolved);
+                        if (r < 0)
+                                return r;
+                        free_and_replace(*value, resolved);
```

Comment:
no need to use STRV_FOREACH_PAIR() if you want to do the same thing for each key and each value anyway. Just use STRV_FOREACH(), and save a few lines of code